server.go

```diff
@@ -61,39 +68,83 @@ var wsUpgrader = websocket.Upgrader{
 
 func (s *SockchatServer) webSocket(w http.ResponseWriter, r *http.Request) {
 	conn := newSockChatWS(w, r)
-	defer s.ShutConnection(conn)
+	defer s.shutConnection(conn)
 	for {
 		receivedMsg, err := conn.ReadSocketMsg()
 		if err != nil {
 			log.Printf("error occured when listening for ws messages, closing connection %v", err)
-			s.ShutConnection(conn)
+			s.shutConnection(conn)
 			break
 		}
 
 		switch receivedMsg.Action {
 		case CreateAction:
-			s.CreateNewChannel(*receivedMsg, conn)
+			s.createNewChannel(*receivedMsg, conn)
 		case JoinAction:
-			s.JoinChannel(*receivedMsg, conn)
+			s.joinChannel(*receivedMsg, conn)
 		case LeaveAction:
-			s.LeaveChannel(*receivedMsg, conn)
+			s.leaveChannel(*receivedMsg, conn)
 		case SendMessageAction:
-			s.SendMessageToChannel(*receivedMsg, conn)
+			s.sendMessageToChannel(*receivedMsg, conn)
 		default:
 			log.Printf("Unexpected request received: %s", receivedMsg.Action)
 			conn.WriteSocketMsg(NewErrorMessage("action not supported"))
 		}
 
 	}
+}
+
+func (s *SockchatServer) register(w http.ResponseWriter, r *http.Request) {
+	userData := UserRequest{}
+	bodyBytes, err := io.ReadAll(r.Body)
+	if err != nil {
+		log.Printf("error reading register request: %v", err)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	err = json.Unmarshal(bodyBytes, &userData)
+	if err != nil {
+		log.Printf("error unmarshaling register request: %v", err)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	err = s.userService.CreateUser(context.TODO(), &userData)
```

Comment:
```suggestion
	ctx, cancel := context.WithTimeout(r.Context(), 3 * time.Second)
	defer cancel()
	err = s.userService.CreateUser(ctx, &userData)
```
Context from request is already available here and could be used to set timeout